### ISSUE TITLE
Nullable analysis fails to understand complicated is (pattern) expression

### ISSUE THREAD

**Issue by pdelvo** [dotnet/roslyn#31881]
**Version Used**: Latest supported by SharpLap and VS 2019 Preview 1.1

**Steps to Reproduce**:

```csharp
using System;
#nullable enable

public class C
{
    public object? AProperty { get; set; }
    public void M(C? input)
    {
    	if (input?.AProperty is string str)
        {
            Console.WriteLine(str.ToString());
            
            // The compiler produces a "warning CS8602: Possible dereference of a null reference." for all the following three lines:
            Console.WriteLine(input?.AProperty.ToString());
            Console.WriteLine(input.AProperty.ToString());
            Console.WriteLine(input.ToString());
        }
    }
}
```
https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQQCYEsYHsEDGuAdmFgOapbGpEC2ADlgDYQIA0MIAbm1mAE92GEAGoAPgAEADAAJJARgDcAWABQAYmJxmzKMFayIxfa3XrJAZnkAmWQGF1Ab3UBIK7NzAAVhAIwAflkAQQAFBFwGNhgBWSdZcggYJVlkJJSAX3VZHPlrSQAWWQBZAAp7IOoGOBgASmzclzVc2Vd+WVKqmoCAOjCIqIQY2SxkeQU5ZBgEeuaWxob51sUATlKphB6AFVwAZWnqclLa2tU5pfcFNa7AvvDI6IFtvYPiI5OzpZzL6+JqmDuA0ez32CEOx1Oi3mP06fxqINe70h53mWTUrjRGSAA==

Note that changing the is pattern to a normal is expression does not make a difference.
**Expected Behavior**:
No warning is produced.
**Actual Behavior**:
The compiler emits three CS8602 warnings in the code.

Because of the if statement we know that input?.AProperty is not null. This implies that both input and input.AProperty are not null. So none on these statements should produce a warning.
